### DISCUSSION TITLE
Random thread id

### DISCUSSION THREAD

**Discussion by omidMolaverdi** [cmgmyr/laravel-messenger#381] [Q&A] [answered]
Hi and thanks for the fantastic work.

I have the questions below:

1. Is there any way that I assign a random id int to the thread in the 
```
$thread = Thread::create([
            'subject' => $input['subject'],
             'id' => rand()    <----------- This does not work
        ]);
```

Thanks very much.

**Comment by omidMolaverdi** [comment]
@cmgmyr can you kindly contribute to this please?

**Comment by cmgmyr** [comment, marked as answer]
Hi @omidMolaverdi, I would not mess around with the auto-incrementing id. If you're looking to obfuscate the IDs used in the URL, I'd add a UUID column to the `threads` table/model, and probably to the `messages` and `participants` too. Then you want to use these new UUIDs for the URLs. This will completely hide the auto-incrementing ids from the users and make it extremely hard to guess.

You'll also want to write your own authorization code to make sure the current user can view a message. For example, if the user isn't in the thread participants, then you want to redirect them somewhere else or show an error.

These two options are not in the package to be more flexible and it's mostly up to the end developer to implement authorization.

**Reply by omidMolaverdi** [reply]
Thanks.
